debezium-core/src/main/java/io/debezium/pipeline/CommonOffsetContext.java

```diff
@@ -5,20 +5,35 @@
  */
 package io.debezium.pipeline;
 
+import java.util.Optional;
+
 import org.apache.kafka.connect.data.Struct;
 
 import io.debezium.connector.SnapshotRecord;
+import io.debezium.connector.SnapshotType;
 import io.debezium.connector.common.BaseSourceInfo;
 import io.debezium.pipeline.spi.OffsetContext;
 
 public abstract class CommonOffsetContext<T extends BaseSourceInfo> implements OffsetContext {
 
+    public static final String SNAPSHOT_COMPLETED_KEY = "snapshot_completed";
+
     protected final T sourceInfo;
+    protected SnapshotType snapshot;
```

Comment:
Could we add a JavaDoc that will state what values will be for `snapshot` and `snapshotCompleted` fields in case of the following scenarios

* incomplete initial snapshot
* completed initial snapshot
* incmplete blocking snapshot
* completed blocking snapshot
* running incremental snapshot
* completed incremental snapshot

Such documentation would work as a contract against which we can assert tests and define code behaviour.